webview/src/experiments/components/table/Indicators.tsx

```diff
@@ -3,51 +3,97 @@ import { SortDefinition } from 'dvc/src/experiments/model/sortBy'
 import cx from 'classnames'
 import { MessageFromWebviewType } from 'dvc/src/webview/contract'
 import { FilteredCounts } from 'dvc/src/experiments/model/filterBy/collect'
+import { TippyProps } from '@tippyjs/react'
 import styles from './styles.module.scss'
 import { Icon } from '../../../shared/components/Icon'
 import SvgSortPrecedence from '../../../shared/components/icons/SortPrecedence'
 import SvgFilter from '../../../shared/components/icons/Filter'
 import { sendMessage } from '../../../shared/vscode'
 import Tooltip from '../../../shared/components/tooltip/Tooltip'
 import tooltipStyles from '../../../shared/components/tooltip/styles.module.scss'
+import { pluralize } from '../../../util/strings'
 
-const Indicator = ({
+export type IndicatorTooltipProps = Pick<TippyProps, 'children'> & {
+  tooltipContent: ReactNode
+}
+
+export const IndicatorTooltip: React.FC<IndicatorTooltipProps> = ({
+  children,
+  tooltipContent
+}) => {
+  const wrapperRef = React.useRef<HTMLDivElement>(null)
+  return (
+    <Tooltip
+      placement="bottom-start"
+      disabled={!tooltipContent}
+      content={tooltipContent}
+      className={tooltipStyles.padded}
+      ref={wrapperRef}
+    >
+      {children}
+    </Tooltip>
+  )
+}
+
+export type CounterBadgeProps = {
+  count?: number
+}
+
+export const CounterBadge: React.FC<CounterBadgeProps> = ({ count }) => {
+  return count ? (
+    <span
+      className={styles.indicatorCount}
+      role={'marquee'}
+      aria-label={`${count ?? 0}`}
+    >
+      {count ?? 0}
```

Comment:
[Q] How is `??` needed here when it can only be a truthy value?. Could you use a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) here?